py/frequenz/client/weather/_types.py

```diff
@@ -243,12 +255,13 @@ def to_ndarray_vlf(
                     array_f_index = 0
 
                     for f_index in feature_indexes:
-                        array[array_t_index, array_l_index, array_f_index] = (
-                            self._forecasts_pb.location_forecasts[l_index]
-                            .forecasts[t_index]
-                            .features[f_index]
-                            .value
-                        )
+                        if l_index >= 0 and t_index >= 0 and f_index >= 0:
```

Comment:
If a requested key was not found we append a `-1`  to the indices (see above) and here we want to skip any keys that were not found. In that case we don't update the position in the array and it will be `NaN` as initialized. Will add a comment here.

Reply:
Updated with additional comments.